hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCache.java

```diff
@@ -135,16 +138,31 @@ public boolean remove(K key) {
     }
 
     @Override
+    public boolean invalidate(K key) {
+        nearCacheStats.incrementInvalidations();
+        return remove(key);
+    }
+
+    @Override
+    public void invalidateAll() {
+        int size = nearCacheRecordStore.size();
```

Comment:
I'd rather have near cache stats updated either at the `NearCache` or at the `NearCacheRecordStore` level, instead of making stats update a mixed concern of both layers. What do you think?
One more related thing: in HD implementation `size()` iterates and sums over segments' size so is prone to races. Maybe this block should be guarded as a whole under the HD record store's lock?